### PR TITLE
drop support for node 16 in preparation for v5 release

### PR DESCRIPTION
**Pull request by khanayan123**
### What does this PR do?
drops support for node 16

### Motivation
v5 release of the tracer

### CODE REVIEW

File: .github/workflows/appsec.yml [hunk 1]
```diff
@@ -27,9 +27,7 @@ jobs:
       - uses: actions/checkout@v2
       - uses: ./.github/actions/node/setup
       - run: yarn install
-      - uses: ./.github/actions/node/16
-      - run: yarn test:appsec:ci
-      - uses: ./.github/actions/node/18
+      - uses: ./.github/actions/node/oldest
```
Comment (Qard) [inline .github/workflows/appsec.yml:30]: Should probably add 20 in there to correspond with the shift of 18 to oldest.

File: .github/workflows/plugins.yml [hunk 2]
```diff
@@ -88,18 +88,31 @@ jobs:
       - uses: actions/checkout@v2
       - uses: ./.github/actions/testagent/start
       - uses: ./.github/actions/node/setup
-      - run: yarn install --ignore-engines
+      - id: pkg
+        run: |
+          content=`cat ./package.json | tr '\n' ' '`
+          echo "::set-output name=json::$content"
```
Comment (Qard) [inline .github/workflows/plugins.yml:94]: Just an FYI that the `set-output` command is [deprecated](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/). I've removed it in #3907 so should probably avoid adding it back again here.

File: .github/workflows/project.yml [hunk 3]
```diff
@@ -18,7 +18,7 @@ jobs:
       # setting fail-fast to false in an attempt to prevent this from happening
       fail-fast: false
       matrix:
-        version: [16, 18, latest]
+        version: [18, latest]
```
Comment (Qard) [inline .github/workflows/project.yml:21]: Should add 20.

**Review by rochdev — APPROVED** [review]

**Review by rochdev — APPROVED** [review]

File: .github/workflows/plugins.yml [hunk 4]
```diff
@@ -1147,7 +1156,7 @@ jobs:
       - run: yarn install
       - uses: ./.github/actions/node/oldest
       - run: yarn test:plugins:ci
-      - uses: ./.github/actions/node/16
+      - uses: ./.github/actions/node/latest
```
Comment (simon-id) [inline .github/workflows/plugins.yml:1159]: bit weird to change 16 to latest when everywhere else in the PR you're changing 16 to oldest/18. Is this intended ?

Comment (Qard) [reply .github/workflows/plugins.yml:1159]: We went from testing only 16 and older to testing `oldest` to `latest`. I know restify has a focus on legacy, but as far as I'm aware it _should_ support all the way up to latest so I think it was just an oversight that the range before this change wasn't testing up to latest.

File: .github/workflows/plugins.yml [hunk 5]
```diff
@@ -89,18 +89,31 @@ jobs:
...
+      - id: pkg
+        run: |
+          content=`cat ./package.json | tr '\n' ' '`
+          echo "json=$content" >> $GITHUB_OUTPUT
+      - id: extract
+        run: |
+          version="${{fromJson(steps.pkg.outputs.json).version}}"
+          majorVersion=$(echo "$version" | cut -d '.' -f 1)
+          echo "Major Version: $majorVersion"
+          echo "MAJOR_VERSION=$majorVersion" >> $GITHUB_ENV
```
Comment (Qard) [inline .github/workflows/plugins.yml:101]: These could probably be combined to one step.

**Review by Qard — APPROVED** [review]
LGTM, just one small suggestion.

**Review by Qard — APPROVED** [review]

**Review by simon-id — DISMISSED** [review]

File: .gitlab/benchmarks.yml [hunk 6]
```diff
@@ -55,12 +55,6 @@ benchmark:
   extends: .benchmarks
   parallel:
     matrix:
-      - MAJOR_VERSION: 16
-        GROUP: 1
-      - MAJOR_VERSION: 16
-        GROUP: 2
-      - MAJOR_VERSION: 16
-        GROUP: 3
       - MAJOR_VERSION: 18
```
Comment (tlhunter) [inline .gitlab/benchmarks.yml:58]: I can add the 20 series in a follow up PR

**Review by tlhunter — APPROVED** [review]

**Review by simon-id — APPROVED** [review]

**Review by Qard — APPROVED** [review]

**Review by szegedi — APPROVED** [review]
🚢 from profiling

**Comment by DarcyRaynerDD** [comment]
From a Serverless perspective, we are locked into supporting node 16 until at least June 2024. This is similar to what happened with node 14 support, where we were locked to using the v3 releases. Are we just dropping support because node 16 is no longer LTS? 
https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html
Is this

**Comment by khanayan123** [comment]
> From a Serverless perspective, we are locked into supporting node 16 until at least June 2024. This is similar to what happened with node 14 support, where we were locked to using the v3 releases. Are we just dropping support because node 16 is no longer LTS? https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html Is this

Yes, ideally this release was supposed to happen a few months ago when Node 16 was officially no longer LTS

**Review by duncanista — APPROVED** [review]
Approving to not block the `v5` release, everything looks good on the Serverless side, but we'd be pinned to the `v4` line until August 2024 – date where AWS will drop support for Node 16 AWS Lambdas.

**Pull request merged**